admin_guide/high_availability.adoc

```diff
@@ -11,37 +11,53 @@
+
+When using VIPs to access a pod with host networking (_e.g. a router_), the application pod should be running on all nodes that are running the ipfailover pods. This enables any of the ipfailover nodes to become the master and service the VIPs when needed. If application pods are not running on all nodes with ipfailover, either some ipfailover nodes will never service the VIPs or some application pods will never receive any traffic. Use the same selector and replication count, for both ipfailover and the application pods, to avoid this mismatch.
 
-IMPORTANT: Even though
-a service is highly available, performance can still be affected.
+While using VIPs to access a service, any of the nodes can be in the ipfailover set of nodes, since the service is reachable on all nodes (no matter where the application pod is running). Any of the ipfailover nodes can become master at any time. The service can either use externalIPs and a service port or it can use a nodePort.
 
-Use cases for high-availability include:
+When using externalIPs in the service definition the VIPs are set to the externalIPs and the ipfailover monitoring port is set to the service port.  A nodePort is open on every node in the cluster and the service will load balance traffic from whatever node currently supports the VIP. In this case the ipfailover monitoring port is set to the nodePort in the service definition.
 
-* I want my cluster to be assigned a resource set and I want the cluster to automatically manage those resources.
-* I want my cluster to be assigned a set of VIPs that the cluster manages and migrates (with zero or minimal downtime) on failure conditions, and I should not be required to perform any manual interactions to update the upstream "discovery" sources (e.g., DNS). The cluster should service all the assigned VIPs when at least a single node is available, despite the current available resources not being sufficient to reach the desired state.
+[IMPORTANT]
+====
+Setting up a nodePort is a privileged operation.
+====
 
+[IMPORTANT]
+====
+Even though a service VIP is highly available, performance can still be affected. *keepalived* makes sure that each of the VIPs is serviced by some node in the configuration, and several VIPs may end up on the same node even when other nodes have none. Strategies that externally load balance across a set of VIPs
+may be thawed when ipfailover puts multiple VIPs on the same node.
+====
+
+When you use ingressIP you can set up ipfailover to have the same VIP range, as the ingressIP range. You can also disable the
+monitoring port. In this case all the VIPs will appear on same node in the cluster. Any user can set up a service with a ingressIP and have it highly available.
+
+[IMPORTANT]
+====
+There are a maximum of 255 VIPs in the cluster.
+====
+
+If you are using the maru thing, then a cluster administrator can set up an ipfailover for the same range but disable the watch port.
```

Comment:
Oops! Forgot to delete the sentence. The discussion should be in the neighboring prose.